### PR TITLE
Feature Wbooks

### PR DESCRIPTION
**Pull request by FayberRueda**
## Summary
- Convert the book cover into a webcomponent with Lit and reuse it in the list.
- Display each of the workbooks listed in a JSON

## Screenshot

Captura de Pantalla 2022-09-30 a la(s) 6 56 51 p m

## Trello card

https://trello.com/c/SZnP55z3

### CODE REVIEW

File: components/book-finder.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          type="text"
+          name="finder"
+          id="inputSearch"
+          placeholder="Busca por título del libro..."
+        />
+      </div>
+    `;
+  }
+}
+customElements.define("book-finder", BookFinder);
```
Comment (dsandoval11) [inline components/book-finder.js:22]: add an EOF line

File: components/book-list.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { LitElement, html } from "lit";
+import styles from "./styles/book-list-styles.js";
+
+
+export class BookList extends LitElement {
```
Comment (dsandoval11) [inline components/book-list.js:5]: Don't export the class

File: components/book-list.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        let element = this.generalRepository[item][key];
+        characters.push({
+          keyB: key,
+          cover: element.cover,
+          autor: element.autor,
+          title: element.title,
+        });
+      }
+    }
+    this.repository = characters;
```
Comment (dsandoval11) [inline components/book-list.js:49]: You can use the map function

File: components/book-list.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  get dataTemplate() {
+    this._dataFormat();
```
Comment (dsandoval11) [inline components/book-list.js:53]: it's not good to call functions inside the render function, you can better define `firstUpdate` or call it in the constructor

File: components/styles/book-finder-styles.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { css } from "lit";
+
+export default css`
+  :host {
+    display: block;
+    font-family: Roboto;
+    margin: 0px;
```
Comment (dsandoval11) [inline components/styles/book-finder-styles.js:7]:
```suggestion
    margin: 0;
```

File: components/styles/book-finder-styles.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  .finder-container {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 70px;
+    width: 100%;
+    margin-bottom: 20px;
+  }
+
+  #inputSearch {
```
Comment (dsandoval11) [inline components/styles/book-finder-styles.js:19]: defines a class. use kebab-case

File: infoBook.json [hunk 7]
```diff
@@ -0,0 +1,114 @@
+[
+  {
+    "book1": {
```
Comment (dsandoval11) [inline infoBook.json:3]: It isn't necessary to define this property book1, book2 ... you can directly define cover, title and author inside the first object.

File: components/book-list.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    return html`
+      <div class="book-container">
+        ${this.repository.map(
+          (characters) => html`
+            <a href="#" class="book-wrapper">
+              <div>
```
Comment (dsandoval11) [inline components/book-list.js:59]: I think this div is unnecessary 🤔

File: components/book-finder.js (hunk 1 above)
Comment (FayberRueda) [reply components/book-finder.js:22]: Done 👍

File: components/book-list.js (hunk 2 above)
Comment (FayberRueda) [reply components/book-list.js:5]: Done 💯

**Pull request closed without merging**